staging/src/k8s.io/cloud-provider/sample/basic_main.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file should be written by each cloud provider.
+// For an minimal working example, please refer to k8s.io/cloud-provider/sample/basic_main.go
+// For an advanced example, please refer to k8s.io/cloud-provider/sample/advanced_main.go
+// For more details, please refer to k8s.io/kubernetes/cmd/cloud-controller-manager/main.go
+
+package sample
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/spf13/pflag"
+	"k8s.io/cloud-provider"
+	"k8s.io/cloud-provider/app"
+	"k8s.io/cloud-provider/options"
+	"k8s.io/component-base/cli/flag"
+	"k8s.io/component-base/logs"
+	_ "k8s.io/component-base/metrics/prometheus/clientgo" // load all the prometheus client-go plugins
+	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
+	"k8s.io/klog/v2"
+)
+
+const (
+	// sampleCloudProviderName shows an sample of using hard coded parameter for CloudProviderName
```

Comment:
Add instructions in the comment for the cloud provider to replace these values?

Reply:
Added more comments